view/frontend/web/js/view/payment/method-renderer/adyen-apple-pay-method.js

```diff
@@ -0,0 +1,184 @@
+
+                    promise.then(function (success) {
+                        var status;
+                        if (success)
+                            status = ApplePaySession.STATUS_SUCCESS;
+                        else
+                            status = ApplePaySession.STATUS_FAILURE;
+
+                        session.completePayment(status);
+
+                        if (success) {
+                            window.location.replace(url.build(window.checkoutConfig.payment[quote.paymentMethod().method].redirectUrl));
+                        }
+                    }, function (reason) {
+                        if (reason.message == "ERROR BILLING") {
+                            var status = session.STATUS_INVALID_BILLING_POSTAL_ADDRESS;
+                        } else if (reason.message == "ERROR SHIPPING") {
+                            var status = session.STATUS_INVALID_SHIPPING_POSTAL_ADDRESS;
+                        } else {
+                            var status = session.STATUS_FAILURE;
+                        }
+                        session.completePayment(status);
+                    });
+                }
+
+                session.begin();
+            },
+            getControllerName: function () {
+                return window.checkoutConfig.payment.iframe.controllerName[this.getCode()];
+            },
+            getPlaceOrderUrl: function () {
+                return window.checkoutConfig.payment.iframe.placeOrderUrl[this.getCode()];
+            },
+            context: function () {
+                return this;
+            },
+            validate: function () {
+                return true;
+            },
+            showLogo: function () {
+                return window.checkoutConfig.payment.adyen.showLogo;
+            },
+            isApplePayAllowed: function () {
+                if (window.ApplePaySession) {
+                    return true;
+                }
+                return false;
+            },
+            performValidation: function (validationURL) {
+                // Return a new promise.
+                return new Promise(function (resolve, reject) {
+
+                    // retrieve payment methods
+                    var serviceUrl = urlBuilder.createUrl('/adyen/request-merchant-session', {});
+
+                    storage.post(
+                        serviceUrl, JSON.stringify('{}')
+                    ).done(
+                        function (response) {
+                            var data = JSON.parse(response);
+                            resolve(data);
+                        }
+                    ).fail(function (error) {
+                        console.log(JSON.stringify(error));
+                        reject(Error("Network Error"));
+                    });
+                });
+            },
+            sendPayment: function (payment, data) {
+                return new Promise(function (resolve, reject) {
```

Comment:
since jquery.when already returns a promise, do we need to wrap it to another promise?